core-java-modules/core-java-calculating-moving-averages/src/test/java/com/baeldung/MovingAverageWithApacheCommonsMathUnitTest.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class MovingAverageWithApacheCommonsMathUnitTest {
+
+    @Test
+    public void when_initial_average_is_calculated_it_should_be_NAN() {
+        MovingAverageWithApacheCommonsMath movingAverageCalculator = new MovingAverageWithApacheCommonsMath(5);
+        assertEquals(Double.NaN, movingAverageCalculator.getMovingAverage(), 0.001);
+    }
+
+    @Test
+    public void when_values_are_added_average_should_update_correctly() {
+        MovingAverageWithApacheCommonsMath movingAverageCalculator = new MovingAverageWithApacheCommonsMath(3);
+        movingAverageCalculator.add(10);
+        assertEquals(10.0, movingAverageCalculator.getMovingAverage(), 0.001);
+        movingAverageCalculator.add(20);
+        assertEquals(15.0, movingAverageCalculator.getMovingAverage(), 0.001);
+        movingAverageCalculator.add(30);
+        assertEquals(20.0, movingAverageCalculator.getMovingAverage(), 0.001);
+    }
+
+    @Test
+    public void when_window_size_is_full_oldest_value_should_be_dropped() {
```

Comment:
let's remove the "_" in every word. Please make it:

whenWindowSizeIsFull_shouldDropTheOldestValue()

Reply:
Fixed.